typescript/sdk/src/core/HyperlaneCore.ts

```diff
@@ -207,27 +292,39 @@ export class HyperlaneCore extends HyperlaneApp<CoreFactories> {
 
   // Redundant with static method but keeping for backwards compatibility
   getDispatchedMessages(
-    sourceTx: ethers.ContractReceipt | ViemTxReceipt,
+    sourceTx: TransactionReceipt | ViemTxReceipt,
   ): DispatchedMessage[] {
     return HyperlaneCore.getDispatchedMessages(sourceTx);
   }
 
   async getDispatchTx(
     originChain: ChainName,
     messageId: string,
-  ): Promise<ethers.ContractReceipt | ViemTxReceipt> {
+    blockNumber?: number,
```

Comment:
Probably out of scope for this PR, but at some point it would be useful to have an indexing abstraction in typescript to search over a range